extensions/amp-lightbox-gallery/1.0/test/test-amp-lightbox-gallery.js

```diff
@@ -0,0 +1,212 @@
+        expect(element.hasAttribute('hidden')).to.be.true;
+
+        element.setAttribute('open', '');
+        await waitFor(
+          () => !element.hasAttribute('hidden'),
+          'lightbox is shown'
+        );
+
+        const renderedImgs = element.shadowRoot.querySelectorAll('img');
+        expect(renderedImgs).to.have.lengthOf(1);
+        expect(renderedImgs[0].tagName).to.equal('IMG');
+        expect(renderedImgs[0].srcset).to.equal('img.jpg 1x');
+
+        await whenCalled(element.setAsContainerInternal);
+        const scroller = element.shadowRoot.querySelector('[part=scroller]');
+        expect(scroller).to.exist;
+        expect(element.setAsContainerInternal).to.be.calledOnce;
+        expect(element.setAsContainerInternal).to.be.calledWith(scroller);
+        expect(element.removeAsContainerInternal).to.not.be.called;
+
+        element.removeAttribute('open');
+        // Extend timeout due to animation delay.
+        await poll(
+          'lightbox is hidden',
+          () => element.hasAttribute('hidden'),
+          undefined,
+          500
+        );
+        expect(element.shadowRoot.querySelector('[part=lightbox]')).to.be.null;
+
+        expect(element.setAsContainerInternal).to.be.calledOnce;
+        expect(element.removeAsContainerInternal).to.be.calledOnce;
+      });
+    });
+
+    describe('imperative api', () => {
+      let element, img;
+
+      beforeEach(async () => {
+        img = html` <img lightbox src="img.jpg" /> `;
+        doc.body.appendChild(img);
+        await installLightboxGallery(env.ampdoc);
+        element = doc.getElementById(TAG);
+        await element.buildInternal();
+      });
+
+      function invocation(method, args = {}) {
+        const source = null;
+        const caller = null;
+        const event = null;
+        const trust = ActionTrust.HIGH;
+        return new ActionInvocation(
+          element,
+          method,
+          args,
+          source,
+          caller,
+          event,
+          trust
+        );
+      }
+
+      it('should open with default action', async () => {
+        env.sandbox.stub(element, 'setAsContainerInternal');
+        env.sandbox.stub(element, 'removeAsContainerInternal');
+
+        expect(element.hasAttribute('open')).to.be.false;
+        expect(element.hasAttribute('hidden')).to.be.true;
+
+        element.enqueAction(invocation(DEFAULT_ACTION));
+        await waitForOpen(element, true);
+        expect(element.hasAttribute('hidden')).to.be.false;
+
+        const renderedImgs = element.shadowRoot.querySelectorAll('img');
+        expect(renderedImgs).to.have.lengthOf(1);
+        expect(renderedImgs[0].tagName).to.equal('IMG');
+        expect(renderedImgs[0].srcset).to.equal('img.jpg 1x');
+
+        await whenCalled(element.setAsContainerInternal);
+        const scroller = element.shadowRoot.querySelector('[part=scroller]');
+        expect(scroller).to.exist;
```

Comment:
nit: as per [Chai's doc on exist](https://www.chaijs.com/api/bdd/#method_exist) it's better to assert the value (or lack thereof). This can probably be changed to `expect(scroller).not.to.be.null` since `querySelector` returns null if element is not found.

Reply:
Great idea, thanks!